### PR TITLE
Feature/review list create

### PR DESCRIPTION
**Pull request by annashamray**
fixes https://github.com/GemeenteUtrecht/record-management-app/issues/8

### CODE REVIEW

File: src/rma/destruction/views.py [hunk 1]
```diff
@@ -99,3 +119,98 @@ def get_queryset(self):
...
+                        field.name: int(field.value())
+                        for field in formset.management_form
+                    },
+                },
+            }
+        )
+        return context
+
+    def form_valid(self, form):
+        if "approve" in self.request.POST:
```
Comment (sergei-maertens) [inline src/rma/destruction/views.py:183]: that's the name of the button, right? What if you set the `name` of the button to `status` and the value to the matching choice option for the model field? Then you don't have to override `form_valid` at all.

File: src/rma/destruction/views.py [hunk 2]
```diff
@@ -99,3 +119,98 @@ def get_queryset(self):
...
+            form.instance.status = ReviewStatus.approved
+
+        return super().form_valid(form)
+
+    def forms_valid(self, form, inlines):
+        response = super().forms_valid(form, inlines)
```
Comment (sergei-maertens) [inline src/rma/destruction/views.py:189]:
```suggestion
    @transaction.atomic
    def forms_valid(self, form, inlines):
        response = super().forms_valid(form, inlines)
```

Important to wrap all this in a transaction, and only call the celery task in `transaction.on_commit`

File: src/rma/destruction/views.py [hunk 3]
```diff
@@ -99,3 +119,98 @@ def get_queryset(self):
...
+            self.request,
+            list_review,
+            template="destruction/logs/review_created.txt",
+            n_items=list_review.item_reviews.count(),
+        )
+        # send notification
+        Notification.objects.create(
+            destruction_list=list_review.destruction_list,
+            user=list_review.destruction_list.author,
+            message=f"Destruction list has been reviewed by {list_review.author}",
```
Comment (sergei-maertens) [inline src/rma/destruction/views.py:203]: should be marked translatable, and please note the correct variable interpolation format for tanslations :-) (f-strings do not work for that!)

File: src/rma/js/components/review-list/review-item-form.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    const name_prefix = (field) => `${prefix}-${index}-${field}`;
+
+    const [ comment, setComment ] = useState("");
+    const suggestion = suggestions[index];
+    const disabled = !!suggestion;
+
+    const actionIcon = (suggestion === "remove" ? "clear": (suggestion === "change_and_remove" ? "create" : "done"));
+
+    // modal
+    const canOpen = zaakDetailPermission === "True";
```
Comment (sergei-maertens) [inline src/rma/js/components/review-list/review-item-form.js:24]: Booleans exist in Javascript :-) why is this not an actual boolean?

File: src/rma/destruction/views.py (hunk 3 above)
Comment (annashamray) [reply src/rma/destruction/views.py]: added using named-string interpolation syntax

File: src/rma/js/components/review-list/review-item-form.js (hunk 4 above)
Comment (annashamray) [reply src/rma/js/components/review-list/review-item-form.js:24]: The value is received from data-* attribute. I thought that they can be only strings. 
https://github.com/GemeenteUtrecht/record-management-app/blob/9782da9da31745dfcd184304df3c3db020e6bbb9/src/rma/js/components/review-list/index.js#L14

File: src/rma/destruction/views.py (hunk 2 above)
Comment (annashamray) [reply src/rma/destruction/views.py:175]: done

File: src/rma/destruction/views.py [hunk 5]
```diff
@@ -99,3 +112,92 @@ def get_queryset(self):
...
+        )
+        # send notification
+        Notification.objects.create(
+            destruction_list=list_review.destruction_list,
+            user=list_review.destruction_list.author,
+            message=_("Destruction list has been reviewed by %(author)s")
+            % {"author": list_review.author},
+        )
+
+        transaction.on_commit(self.process_destruction_list(list_review))
```
Comment (sergei-maertens) [inline src/rma/destruction/views.py:193]:
```suggestion
        transaction.on_commit(lambda: self.process_destruction_list(list_review))
```

File: src/rma/js/components/review-list/review-item-form.js (hunk 4 above)
Comment (sergei-maertens) [reply src/rma/js/components/review-list/review-item-form.js:24]: yeah, you'd have to cast that explicitly... I'll leave it for now.

File: src/rma/destruction/views.py (hunk 5 above)
Comment (annashamray) [reply src/rma/destruction/views.py]: thanks for fixing it!

**Pull request merged**